cmd/boost/deal_cmd.go

```diff
@@ -0,0 +1,344 @@
+		if err != nil {
+			return fmt.Errorf("failed to create a deal proposal: %w", err)
+		}
+
+		dealParams := types.DealParams{
+			DealUUID:           dealUuid,
+			ClientDealProposal: *dealProposal,
+			DealDataRoot:       rootCid,
+			Transfer: types.Transfer{
+				Type:   "http",
+				Params: paramsBytes,
+				Size:   carFileSize,
+			},
+		}
+
+		log.Debugw("about to submit deal proposal", "uuid", dealUuid.String())
+
+		var resp types.DealResponse
+		if err := doRpc(ctx, s, &dealParams, &resp); err != nil {
+			return fmt.Errorf("send proposal rpc: %w", err)
+		}
+
+		if !resp.Accepted {
+			return fmt.Errorf("deal proposal rejected: %s", resp.Message)
+		}
+
+		msg := fmt.Sprintf("sent deal proposal %s to storage provider %s:\n", dealUuid, maddr)
+		msg += fmt.Sprintf("  client wallet: %s\n", walletAddr)
+		msg += fmt.Sprintf("  payload cid: %s\n", rootCid)
+		msg += fmt.Sprintf("  url: %s\n", transferParams.URL)
+		msg += fmt.Sprintf("  commp: %s\n", dealProposal.Proposal.PieceCID)
+		msg += fmt.Sprintf("  start epoch: %d\n", dealProposal.Proposal.StartEpoch)
+		msg += fmt.Sprintf("  end epoch: %d\n", dealProposal.Proposal.EndEpoch)
+		msg += fmt.Sprintf("  client collateral: %s\n", chain_types.FIL(dealProposal.Proposal.ClientCollateral).Short())
+		msg += fmt.Sprintf("  provider collateral: %s\n", chain_types.FIL(dealProposal.Proposal.ProviderCollateral).Short())
+		fmt.Println(msg)
+
+		return nil
+	},
+}
+
+func dealProposal(ctx context.Context, n *clinode.Node, rootCid cid.Cid, pieceSize abi.PaddedPieceSize, pieceCid cid.Cid, minerAddr address.Address, startEpoch abi.ChainEpoch, duration int, verified bool, providerCollateral abi.TokenAmount, storagePricePerEpoch abi.TokenAmount) (*market.ClientDealProposal, error) {
+	clientAddr, err := n.Wallet.GetDefault()
+	if err != nil {
+		return nil, err
+	}
+
+	endEpoch := startEpoch + abi.ChainEpoch(duration)
+	proposal := market.DealProposal{
+		PieceCID:             pieceCid,
+		PieceSize:            pieceSize,
+		VerifiedDeal:         verified,
+		Client:               clientAddr,
+		Provider:             minerAddr,
+		Label:                rootCid.String(),
+		StartEpoch:           startEpoch,
+		EndEpoch:             endEpoch,
+		StoragePricePerEpoch: storagePricePerEpoch,
+		ProviderCollateral:   providerCollateral,
+	}
+
+	buf, err := cborutil.Dump(&proposal)
+	if err != nil {
+		return nil, err
+	}
+
+	sig, err := n.Wallet.WalletSign(ctx, clientAddr, buf, api.MsgMeta{Type: api.MTDealProposal})
+	if err != nil {
+		return nil, fmt.Errorf("wallet sign failed: %w", err)
+	}
+
+	return &market.ClientDealProposal{
+		Proposal:        proposal,
+		ClientSignature: *sig,
+	}, nil
+}
+
+func doRpc(ctx context.Context, s inet.Stream, req interface{}, resp interface{}) error {
```

Comment:
If the context is cancelled will the `doRpc` function return? Just want to make sure people can ctrl-c if the command hangs while trying to contact the SP

Reply:
I think this is done now, will test it out.